validator/node/node.go

```diff
@@ -821,15 +818,13 @@ func (c *ValidatorClient) registerRPCGatewayService(router *mux.Router) error {
 		),
 		gwruntime.WithForwardResponseOption(gateway.HttpResponseModifier),
 	)
-	muxHandler := func(apiMware *apimiddleware.ApiProxyMiddleware, h http.HandlerFunc, w http.ResponseWriter, req *http.Request) {
+
+	// TODO: remove api middleware from this
+	muxHandler := func(_ *apimiddleware.ApiProxyMiddleware, h http.HandlerFunc, w http.ResponseWriter, req *http.Request) {
 		// The validator gateway handler requires this special logic as it serves two kinds of APIs, namely
-		// the standard validator keymanager API under the /eth namespace, and the Prysm internal
 		// validator API under the /api namespace. Finally, it also serves requests to host the validator web UI.
-		if strings.HasPrefix(req.URL.Path, "/api/eth/") {
-			req.URL.Path = strings.Replace(req.URL.Path, "/api", "", 1)
-			// If the prefix has /eth/, we handle it with the standard API gateway middleware.
-			apiMware.ServeHTTP(w, req)
-		} else if strings.HasPrefix(req.URL.Path, "/api") {
+		// note: keymanager api endpoints are handled by reregistering all the route handlers before the catchall
```

Comment:
I don't understand this. Can you explain this in more detail?

Reply:
I need to remove this one, instead of reregistering i am doing it immediately when the server is initiated to avoid the race I'll update this.